src/components/PlantingSites/index.tsx

```diff
@@ -41,13 +47,31 @@ export function PlantingSitesWrapper({ reloadTracking }: PlantingSitesProps): JS
   const dispatch = useAppDispatch();
   const trackingV2 = isEnabled('TrackingV2');
 
+  const observationsResults = useAppSelector((state) =>
+    selectPlantingSiteObservationsResults(state, Number(plantingSiteId))
+  );
+  const observationsResultsError = useAppSelector((state) =>
+    selectPlantingSiteObservationsResultsError(state, Number(plantingSiteId))
+  );
+
+  const plantingSites = useAppSelector(selectPlantingSites);
+  const plantingSitesError = useAppSelector(selectPlantingSitesError);
+
   useEffect(() => {
     const siteId = Number(plantingSiteId);
     if (!isNaN(siteId) && trackingV2) {
       dispatch(requestPlantingSiteObservationsResults(selectedOrganization.id, siteId));
     }
   }, [dispatch, selectedOrganization.id, plantingSiteId, trackingV2]);
 
+  // show spinner while initializing data
+  if (
+    (observationsResults === undefined && !observationsResultsError) ||
+    (plantingSites === undefined && !plantingSitesError)
+  ) {
+    return <CircularProgress sx={{ margin: 'auto' }} />;
```

Comment:
lifted this from Observations.. basically spin while we are loading data. this is needed so subsequent views can do error handling with selected data (at this point data was already fetched)